script.py

```diff
@@ -0,0 +1,132 @@
+        'config_url': True,
+        'log_url': True,
+    },
+    'tests': {
+        'output_files': [{'url': True}],
+        'log_url': True,
+    },
+}
+
+
+def extract_fields(spec, data):
+    """
+    Extract values of fields from data according to a specification.
+
+    Args:
+        spec:   The specification of fields to extract the values from.
+        data:   The data to extract the field values from.
+
+    Returns:
+        An array of tuples, where each tuple contains:
+            * A tuple containing dictionary keys leading to the field
+            * The field value
+    """
+    tuples = []
+
+    if spec is True:
+        return [(tuple(), data)]
+    if isinstance(spec, dict) and isinstance(data, dict):
+        for obj_type, obj_spec in spec.items():
+            obj = data.get(obj_type)
+            if obj is not None:
+                inner_tuples = extract_fields(obj_spec, obj)
+                # prefix the path from inner tuples with obj_type
+                for field_path, field_value in inner_tuples:
+                    tuples.append(((obj_type,) + field_path, field_value))
+    elif isinstance(spec, list) and isinstance(data, list):
+        assert len(spec) == 1
+        obj_spec = spec[0]
+        for obj in data:
+            tuples += extract_fields(obj_spec, obj)
+    return tuples
+
+
+def get_url_info(url):
+    """Extract url info and returns fields value."""
+    status_code = 0
+    content_type = ''
+    size = ''
+    extension = os.path.splitext(urlparse(url).path)[1]
+    url_length = sys.getsizeof(url.encode('utf-8'))
+    # Some CI systems send links to index pages, instead of actual files,
+    # which is against the schema, and we don't really want to cache them.
+    # That's why we ignore URLs ending with "/"
+    if url.endswith('/'):
+        return (status_code, content_type, size, extension, url_length)
+    try:
+        response = requests.head(url, timeout=10)
+    except requests.exceptions.RequestException as error:
+        print(f"Error while trying to get URL info for {url}: {error!r}")
+    else:
+        status_code = response.status_code
+
+        if status_code == 200:
+            content_type = response.headers.get('Content-Type')
+            size = response.headers.get('Content-Length')
+
+    return (status_code, content_type, size, extension, url_length)
+
+
+def main():
+    """Extract URL info from JSON data and save it in a CSV file."""
+
+    # Prepare CSV file
+    csv_writer = csv.writer(sys.stdout)
+    csv_writer.writerow([
+        'Origin', 'Field Path', 'URL', 'Status Code',
+        'Content Type', 'File Extension', 'URL Length (Bytes)', 'Size'
+    ])
+
+    url_path_set = set()  # Track unique field path and URL combinations
+
+    # Read JSON file path from the terminal
+    json_file_path = sys.argv[1]
+
+    # Read JSON file
+    with open(json_file_path, 'r', encoding='utf-8') as json_file:
+        data = json.load(json_file)
+
```

Comment:
As a final lint, let's try to make this function closer to 
`Input -> Process -> Output` by moving the .json reading to the top

```suggestion
    # Read JSON file path from the terminal
    json_file_path = sys.argv[1]

    # Read JSON file
    with open(json_file_path, 'r', encoding='utf-8') as json_file:
        data = json.load(json_file)

    # Prepare CSV file
    csv_writer = csv.writer(sys.stdout)
    csv_writer.writerow([
        'Origin', 'Field Path', 'URL', 'Status Code',
        'Content Type', 'File Extension', 'URL Length (Bytes)', 'Size'
    ])

    url_path_set = set()  # Track unique field path and URL combinations
```

This can enhance the feedback from our code whenever there's an error while reading the file.
(Otherwise it would print the CSV header before the error message)

Reply:
Thanks @tales-aparecida Learning lots of new things for arranging a code, I will update it.